src/shared/cloudformation/cloudformation.ts

```diff
@@ -747,3 +760,160 @@ export namespace CloudFormation {
         return s.replace(/[^A-Za-z0-9]/g, '')
     }
 }
+
+let CFN_SCHEMA_PATH = ''
+let SAM_SCHEMA_PATH = ''
+const MANIFEST_URL = 'https://api.github.com/repos/awslabs/goformation/releases/latest'
+
+/**
+ * Loads JSON schemas for CFN and SAM templates.
+ * Checks manifest and downloads new schemas if the manifest version has been bumped.
+ * Uses local, predownloaded version if up-to-date or network call fails
+ * If the user has not previously used the toolkit and cannot pull the manifest, does not provide template autocomplete.
+ * @param extensionContext 
+ */
+export async function refreshSchemas(extensionContext: vscode.ExtensionContext): Promise<void> {
+    CFN_SCHEMA_PATH = normalizeSeparator(path.join(extensionContext.globalStoragePath, 'cloudformation.schema.json'))
+    SAM_SCHEMA_PATH = normalizeSeparator(path.join(extensionContext.globalStoragePath, 'sam.schema.json'))
+    let manifest: string | undefined
+    try {
+        const manifestFetcher = new HttpResourceFetcher(MANIFEST_URL, { showUrl: true })
+        manifest = await manifestFetcher.get()
+    } catch (e) {
+        getLogger().error(`Failed getting manifest at ${MANIFEST_URL}:`, e)
+
+        return
+    }
+
+    if (!manifest) {
+        getLogger().error(`Schema manifest at ${MANIFEST_URL} was undefined`)
+
+        return
+    }
+
+    try {
+        const details = getManifestDetails(manifest)
+
+        await getRemoteOrCachedFile({
+            filepath: CFN_SCHEMA_PATH,
+            version: details.version,
+            url: details.cfnUrl,
+            cacheKey: 'cfnSchemaVersion',
+        })
+        await getRemoteOrCachedFile({
+            filepath: SAM_SCHEMA_PATH,
+            version: details.version,
+            url: details.samUrl,
+            cacheKey: 'samSchemaVersion',
+        })
+    } catch (e) {
+        getLogger().error('Could not get details from manifest:', (e as Error).message)
+    }
+}
+
+/**
+ * Maps a template file to a CFN or SAM schema.
+ * If present, removes association with the other type of schema.
+ * Does not modify other schemas not managed by AWS.
+ * @param path Template file path
+ * @param type Template type to use for filepath
+ */
+export async function updateYamlSchemasArray(
+    path: string,
+    type: 'cfn' | 'sam',
+    config: WorkspaceConfiguration = vscode.workspace.getConfiguration('yaml'),
+    paths: { cfnSchema: string; samSchema: string } = { cfnSchema: CFN_SCHEMA_PATH, samSchema: SAM_SCHEMA_PATH }
```

Comment:
What happens if the schema paths are never initialized, do we just write `''` to the settings?

Reply:
It'll still point to the correct files (these don't have a version number attached and are initialized prior to any network calls): https://github.com/aws/aws-toolkit-vscode/pull/1748/files#diff-784b68df0bd652e4a3e092550c71a4bba57bd8bf8f09e54913e47df032290c4aR776 but the files won't exist. Unsure if that would give the YAML extension fits or not.

They're initialized with a `let` since they need the `extensionContext` value from activation.

Reply:
Ah I see now. Yeah this should be fine, extension might complain but invalid paths should be expected by the extension.

Reply:
On second thought, it definitely shouldn't break it considering how many paths I tried that it didn't like when I was testing the functionality.